controllers/autoneg.go

```diff
@@ -74,10 +79,41 @@ func (b *BackendController) updateBackends(name string, svc *compute.BackendServ
 	// Perform optimistic locking to ensure we patch the intended object version
 	p := compute.NewBackendServicesService(b.s).Patch(b.project, name, svc)
 	p.Header().Set("If-match", svc.Header.Get("ETag"))
-	_, err := p.Do()
+	res, err := p.Do()
+	if err != nil {
+		return err
+	}
+	ctx := context.Background()
```

Comment:
What would you think about having some reasonable timeout here? I might suggest using the same as the terraform provider: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_backend_service#timeouts

I'd prefer not to have autoneg potentially block forever here without returning status to the API server, but I'm open to alternatives. The k8s reconcile process will also retry with backoff.

Reply:
We do have 15 minutes as default time out set here https://github.com/cenkalti/backoff/blob/v4/exponential.go#L80 - would that work?

Reply:
I'd prefer to have it explicitly set here, rather than in a dependency which could change. Make sense?